File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/Slot.ts
import { CodeableConcept } from "./CodeableConcept";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";

export type Slot = DomainResource & {
    schedule: Reference;
    serviceCategory?: CodeableConcept[];
    specialty?: CodeableConcept[];
    start: string;
    serviceType?: CodeableConcept[];
    appointmentType?: CodeableConcept;
    status: string;
    comment?: string;
    identifier?: Identifier[];
    end: string;
    overbooked?: boolean;
};

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/Slot.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import CodeableConcept
from base import DomainResource
from base import Reference
from base import Identifier

@dataclass(kw_only=True)
class Slot(DomainResource):
    schedule: Reference
    service_category: Optional[List[CodeableConcept]] = field(default_factory=list)
    specialty: Optional[List[CodeableConcept]] = field(default_factory=list)
    start: str
    service_type: Optional[List[CodeableConcept]] = field(default_factory=list)
    appointment_type: Optional[CodeableConcept] = None
    status: str
    comment: Optional[str] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    end: str
    overbooked: Optional[bool] = None